kernel driver fix - drm_gem_object_vmap

cherry picked from https://github.com/NVIDIA/open-gpu-kernel-modules
550.67 tag
---
 kernel/conftest.sh | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/kernel/conftest.sh b/kernel/conftest.sh
index 04ebb9a..c5d3880 100755
--- a/kernel/conftest.sh
+++ b/kernel/conftest.sh
@@ -5189,11 +5189,15 @@ compile_test() {
             # commit 49a3f51dfeee ("drm/gem: Use struct dma_buf_map in GEM
             # vmap ops and convert GEM backends") in v5.11.
             #
+            # Note that the 'map' argument type is changed from 'struct dma_buf_map'
+            # to 'struct iosys_map' by commit 7938f4218168 ("dma-buf-map: Rename
+            # to iosys-map) in v5.18.
+            #
             CODE="
             #include <drm/drm_gem.h>
             int conftest_drm_gem_object_vmap_has_map_arg(
-                    struct drm_gem_object *obj, struct dma_buf_map *map) {
-                return obj->funcs->vmap(obj, map);
+                    struct drm_gem_object *obj) {
+                return obj->funcs->vmap(obj, NULL);
             }"
 
             compile_check_conftest "$CODE" "NV_DRM_GEM_OBJECT_VMAP_HAS_MAP_ARG" "" "types"
-- 
2.43.2
